applicationset/utils/utils.go

```diff
@@ -448,6 +450,54 @@ func SanitizeName(name string) string {
 	return strings.Trim(name, "-.")
 }
 
+// SlugifyName generates a URL-friendly slug from the provided name and additional options.
```

Comment:
Could you update the AppSet docs to describe this function and give some example usage?

Reply:
What about somewhere [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/GoTemplate/#introduction:~:text=An%20additional%20normalize%20function%20makes%20any%20string%20parameter%20usable%20as%20a%20valid%20DNS%20name%20by%20replacing%20invalid%20characters%20with%20hyphens%20and%20truncating%20at%20253%20characters.%20This%20is%20useful%20when%20making%20parameters%20safe%20for%20things%20like%20Application%20names.) ?